### PR TITLE
Namespaced datasets plumbing part 2 - added Id.DatasetInstance

### PR DESCRIPTION
**Pull request by bdmogal**
Summary:
1. Adds ``Id.DatasetInstance`` containing *namespaceId + datasetInstanceName*.
2. Uses ``Id.DatasetInstance`` in all methods in ``DatasetFramework``.
3. Ignores *namespaceId* currently, usage will come in a separate PR.

Affects a large number of files, but no functional change other than making *namespaceId* available in every dataset instance yet.

### CODE REVIEW

**Comment by bdmogal** [comment]
Build: http://builds.cask.co/browse/CDAP-DUT781-2

File: cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java [hunk 1]
```diff
@@ -57,6 +58,8 @@
   private static final Gson GSON = new Gson();
   private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
   private static final String PROPERTY_COLUMN = "properties";
+  private static final Id.DatasetInstance configStoreatasetInstanceId =
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java]: rename var

File: cdap-proto/src/main/java/co/cask/cdap/proto/Id.java [hunk 2]
```diff
@@ -514,4 +514,67 @@ public static DatasetModule from(String namespaceId, String moduleId) {
...
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(namespace, instanceId);
+    }
+
+    @Override
+    public String toString() {
+      return Objects.toStringHelper(this)
+        .add("namespace", namespace)
```
Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: I don't this toString is proper (or as expected). Try it out.

(namespaceId doesn't have an overridden toString, so either put that, or do namespace.getIt())

**Comment by anwar6953** [comment]
Besides two notes, remainder lgtm

File: cdap-proto/src/main/java/co/cask/cdap/proto/Id.java (hunk 2 above)
Comment (bdmogal) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: Great catch! Infact, I hit that problem in an upcoming PR. Going to override ``toString()`` in ``Id.Namespace`` to return just the ``String`` ``namespaceId`` that it wraps.

File: cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java (hunk 1 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java]: will do.

**Comment by bdmogal** [comment]
@anwar6953 Addressed comments, please review.

**Comment by anwar6953** [comment]
the rest looked good to me.

**Comment by bdmogal** [comment]
Merging.

**Pull request merged**